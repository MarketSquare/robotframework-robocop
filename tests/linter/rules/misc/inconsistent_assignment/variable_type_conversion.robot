*** Test Cases ***
Assignment
    [Tags]    robot:continue-on-failure
    # In simple cases the VAR syntax is more convenient.
    ${number: int}    Set Variable    42
    Should Be Equal    ${number}    ${42}

    # we can also convert other keywords output
    ${number: str}    Convert To Integer    42
    Should Be Equal    ${number}    ${42}

    # or even convert to list
    ${variable: list[str]}    Set Variable    42    43
    Should Be Equal    ${variable}    ${42}

    # In this example conversion is more useful.
    ${match}    ${version: float} =    Should Match Regexp    RF 7.3    ^RF (\\d+\\.\\d+)$
    Should Be Equal    ${match}      RF 7.3
    Should Be Equal    ${version}    ${7.3}
