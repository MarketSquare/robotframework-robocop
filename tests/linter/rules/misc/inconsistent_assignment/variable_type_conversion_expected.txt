variable_type_conversion.robot:17:17 [W] MISC04 The assignment sign is not consistent within the file. Expected '' but got ' =' instead